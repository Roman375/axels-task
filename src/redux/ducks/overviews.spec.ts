import {
  overview,
  LOAD_OVERVIEW,
  SET_OVERVIEW,
  initialState,
  getOverviews,
  setOverview,
} from './overviews'

describe('Overviews', () => {
  //Snapshot
  describe('With Snapshot Testing', () => {
    it('handles ACTION_TYPE correctly', () => {
      const actual = overview(initialState, SET_OVERVIEW)
      expect(actual).toMatchSnapshot()
    })
  })

  //Reducer
  describe('Reducer testing', () => {
    it('SET_OVERVIEW', () => {
      const action = {
        type: SET_OVERVIEW,
        payload: ['John Hartman', { 'direct-subordinates': ['d', 'f'] }],
      }

      expect(overview(initialState, action)).toEqual([
        ...initialState,
        ...action.payload,
      ])
    })
  })

  //Actions
  describe('Action getOverviews testing', () => {
    it('getOverviews()', () => {
      const employeeName = 'Name'
      expect(getOverviews(employeeName)).toEqual({
        type: LOAD_OVERVIEW,
        employeeName,
      })
    })
  })

  describe('Action SET_OVERVIEW testing', () => {
    const action = {
      type: SET_OVERVIEW,
      payload: ['John Hartman', { 'direct-subordinates': ['d', 'f'] }],
    }

    it('setOverview()', () => {
      expect(setOverview(action.payload)).toEqual(action)
    })
  })
})
