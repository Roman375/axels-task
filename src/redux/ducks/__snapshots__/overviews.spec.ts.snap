// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overviews With Snapshot Testing handles ACTION_TYPE correctly 1`] = `Array []`;
